pilot/pkg/model/service.go

```diff
@@ -822,7 +829,17 @@ func IsValidSubsetKey(s string) bool {
 func ParseSubsetKey(s string) (direction TrafficDirection, subsetName string, hostname Hostname, port int) {
 	parts := strings.Split(s, "|")
 	if len(parts) < 4 {
-		return
+		// This could be the DNS srv form of the cluster that uses outbound_.port_.subset_.hostname
+		// Since we dont want every callsite to implement the logic to differentiate between the two forms
+		// we add an alternate parser here.
+		if !strings.HasPrefix(s, fmt.Sprintf("%s_", TrafficDirectionOutbound)) &&
+			!strings.HasPrefix(s, fmt.Sprintf("%s_", TrafficDirectionInbound)) {
+			return
+		}
+		parts = strings.SplitN(s, ".", 4)
+		parts[0] = strings.TrimSuffix(parts[0], "_")
+		parts[1] = strings.TrimSuffix(parts[1], "_")
+		parts[2] = strings.TrimSuffix(parts[2], "_")
```

Comment:
Would the logic be clearer: 
  if (strings.HasPrefix(s, fmt.Sprintf("%s_", TrafficDirectionOutbound)) ||
 			strings.HasPrefix(s, fmt.Sprintf("%s_", TrafficDirectionInbound) {
          parts = strings.Split(s, ".")
          ......
}  else {
          parts := strings.Split(s, "|")
}